packages/api/src/campaign/campaign.resolver.ts

```diff
@@ -0,0 +1,115 @@
+import { PaginatedResponseFactory, SubjectEntity, validateNotModified } from "@comet/cms-api";
+import { EntityManager, EntityRepository, FindOptions, wrap } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { Type } from "@nestjs/common";
+import { Args, ArgsType, ID, Mutation, ObjectType, Query, Resolver } from "@nestjs/graphql";
+import { CampaignScopeInterface } from "src/types";
+
+import { DynamicDtoValidationPipe } from "../validation/dynamic-dto-validation.pipe";
+import { CampaignsService } from "./campaigns.service";
+import { CampaignArgsFactory } from "./dto/campaign-args.factory";
+import { CampaignInputInterface } from "./dto/campaign-input.factory";
+import { CampaignInterface } from "./entities/campaign-entity.factory";
+
+export function createCampaignsResolver({
+    Campaign,
+    CampaignInput,
+    Scope,
+}: {
+    Campaign: Type<CampaignInterface>;
+    CampaignInput: Type<CampaignInputInterface>;
+    Scope: Type<CampaignScopeInterface>;
+}): Type<unknown> {
+    @ObjectType()
+    class PaginatedCampaigns extends PaginatedResponseFactory.create(Campaign) {}
+
+    @ArgsType()
+    class CampaignsArgs extends CampaignArgsFactory.create({ Scope }) {}
+
+    @Resolver(() => Campaign)
+    class CampaignsResolver {
+        constructor(
+            private readonly campaignsService: CampaignsService,
+            private readonly entityManager: EntityManager,
+            @InjectRepository("Campaign") private readonly repository: EntityRepository<CampaignInterface>,
+        ) {}
+
+        @Query(() => Campaign)
+        @SubjectEntity(Campaign)
+        async campaign(@Args("id", { type: () => ID }) id: string): Promise<CampaignInterface> {
+            const campaign = await this.repository.findOneOrFail(id);
+            return campaign;
+        }
+
+        @Query(() => PaginatedCampaigns)
+        async campaigns(@Args() { search, filter, sort, offset, limit, scope }: CampaignsArgs): Promise<PaginatedCampaigns> {
+            const where = this.campaignsService.getFindCondition({ search, filter });
+
+            (where as any).scope = scope;
```

Comment:
why is this any needed ?

Reply:
It's not needed. I removed it in [8194b6e](https://github.com/vivid-planet/comet-brevo-module/pull/8/commits/8194b6e388683ac8fa128d07e0e9e7108eb2659b)